pkg/controller/job/job_controller.go

```diff
@@ -843,9 +845,15 @@ func (jm *Controller) manageJob(job *batch.Job, activePods []*v1.Pod, succeeded
 					if completionIndex != unknownCompletionIndex {
 						template = podTemplate.DeepCopy()
 						addCompletionIndexAnnotation(template, completionIndex)
+						template.Spec.Hostname = fmt.Sprintf("%s-%d", job.Name, completionIndex)
 					}
 					defer wait.Done()
-					err := jm.podControl.CreatePodsWithControllerRef(job.Namespace, template, job, metav1.NewControllerRef(job, controllerKind))
+					appendIndex := "-" + strconv.Itoa(completionIndex) + "-"
+					generateNamePrefix := job.Name + appendIndex
+					if len(appendIndex) > names.MaxGeneratedNameLength {
```

Comment:
shouldn't this be `len(generateNamePrefix) > names.MaxGeneratedNameLenght`?

And below, you probably need to cut the job name to make space for the index.

Better extract this in a utility function in `indexed_job_utils.go` and add a test for it.

Reply:
We already cut the job name as a part of generateName for make space for the index. Anywhere else needed to cut job name for index?

Reply:
I'm not sure the code is correct for all sizes. Extract the code in a function and add a test.

Reply:
Done.